File: backend/src/main/java/pl/kosmala/shop/order/model/dto/OrderDto.java
package pl.kosmala.shop.order.model.dto;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import lombok.Builder;
import lombok.Getter;

import java.util.List;

@Getter
@Builder
public class OrderDto
{
    @NotBlank
    private String firstname;
    @NotBlank
    private String lastname;
    @NotBlank
    private String street;
    @NotBlank
    private String postal;
    @NotBlank
    private String city;
    @NotBlank
    @Email
    private String email;
    @NotBlank
    private String phone;
    @NotBlank
    private String productslug;
    @NotBlank
    private Long paymentId;
    private List<String> friendEmails;
}

File: frontend/src/app/modules/order/model/OrderDto.ts
export interface OrderDto
{
    firstname: string,
    lastname: string,
    street: string,
    postal: string,
    city: string,
    email: string,
    phone: string,
    transport: string,
    departureCity: string,
    productslug: string,
    paymentId: number,
    friendEmails: string[] 
}